RTN-023.tex

```diff
@@ -50,6 +50,372 @@
+
+	\begin{itemize}
+
+	\item Should be able to 'split' a large set of inputs into
+	serveral managable sized pieces.  For instance, DP0.2 consists of
+	about 40,000 visits, and a typical Rubin night of observing will
+	take 1,000-2,000 science exposures.   So a managable unit for
+	single visit processing might be of size 1000 exposures/visits and
+	a DP0.2 processing would split the 40,000 visits into 1000 visit
+	groups.  Consideration should be given to how long (wallclock time)
+	a single campaign 'unit' may be run and to not generally exceed 
+	24 hours to a few days to avoid interruptions from downtimes.
+	Also depends on ability of system to recover from interruptions and
+	continue without needing to restart the whole campaign from
+	the beginning.
+	
+
+	\item  Should allow(?) 'last minute editing' of configs before
+	submission to tweak something, change a pipeline version or
+	visit input list
+
+	\item each campaign should be identified by some unique id string, 
+	perhaps same as workflow manager identifying string?
+
+	\end{itemize}
+
+	\item Ability to store and manage a set of BPS submit scripts 
+	(or equivalent).
+	For instance, as each BPS submit yaml is performed, the yaml
+	script contents could be saved in a database with metadata such
+	as time of submission, location of physical processing
+	(SLAC, IN2P3, UK, Cloud, user-laptop), current state (non started,
+	running, finished-no-errors, finished-with-errors)
+
+
+	\item Ability to quickly access and view logs of each 
+	campaign's individual tasks/jobs to determine errors
+
+	\item Ability to view overall state of running and recently
+	completed campaigns, without being overwhelmed by long-ago completed
+	or abandoned campaigns.
+
+	\item As runs may be distributed 'world-wide' for execution,
+	local sites should be able to view only those jobs running at
+	their local site on local resources or get a wider 'world-wide' view.
+
+	\item Ability to straight-forwardly dump dataids (or filenames
+	with paths) and metadata (exposureid, filter) of inputs or outputs
+	from some specific job or task of the processing campaign to
+	track down issues and errors.  This ability may be present already
+	but some 'helper' scripts could simplify the process.
+
+	\item Ability to watch, in realtime, as jobs or tasks within
+	a campaign or set of campaigns are running.  There is a tool	
+	called 'bps report' which does this at some level, showing how
+	many individual inputs in a campaign set have been processed
+	so far.
+
+	\end {enumerate}
+
+\section{BPS and the PanDA system}
+
+For DP0 \cite{RTN-013}, Rubin is using 
+the \cite{PanDA} workload data.  The Batch Production 
+Service (BPS) \cite{LDM-636} is used as an interface to PanDA.  
+
+BPS:
+\begin{enumerate}
+
+\item Accepts a yaml formatted keyword-value set of parameters which
+defines a single run. (see example run specification yaml below).
+
+\item After 'bps submit <run-spec.yaml>', queries the butler to generate
```

Comment:
I don't think this sentence ends. I'm confused by the first comma and the second comma seems to be describing what a quantum graph is as a bracketed aside.